source/s3_auto_ranged_put.c

```diff
@@ -14,110 +19,37 @@ static const struct aws_byte_cursor s_create_multipart_upload_copy_headers[] = {
-
-static void s_s3_auto_ranged_put_cancel_finished(struct aws_s3_meta_request *meta_request);
-
 static void s_s3_meta_request_auto_ranged_put_destroy(struct aws_s3_meta_request *meta_request);
 
-static int s_s3_auto_ranged_put_next_request(
+static void s_s3_auto_ranged_put_update(
     struct aws_s3_meta_request *meta_request,
-    struct aws_s3_request **out_request);
+    uint32_t flags,
+    struct aws_s3_request **out_request,
+    enum aws_s3_meta_request_update_status *out_status);
 
 static int s_s3_auto_ranged_put_prepare_request(
     struct aws_s3_meta_request *meta_request,
     struct aws_s3_client *client,
     struct aws_s3_vip_connection *vip_connection,
     bool is_initial_prepare);
 
-static int s_s3_auto_ranged_put_header_block_done(
-    struct aws_http_stream *stream,
-    enum aws_http_header_block header_block,
-    struct aws_s3_vip_connection *vip_connection);
-
-static int s_s3_auto_ranged_put_stream_complete(
-    struct aws_http_stream *stream,
-    struct aws_s3_vip_connection *vip_connection);
-
-static void s_s3_auto_ranged_put_finish(
+static void s_s3_auto_ranged_put_request_finished(
     struct aws_s3_meta_request *meta_request,
-    const struct aws_s3_meta_request_finish_options *finish_options);
-
-static void s_s3_auto_ranged_put_notify_request_destroyed(
-    struct aws_s3_meta_request *meta_request,
-    struct aws_s3_request *request);
+    struct aws_s3_request *request,
+    int error_code);
 
 static struct aws_s3_meta_request_vtable s_s3_auto_ranged_put_vtable = {
-    .next_request = s_s3_auto_ranged_put_next_request,
+    .update = s_s3_auto_ranged_put_update,
     .send_request_finish = aws_s3_meta_request_send_request_finish_default,
     .prepare_request = s_s3_auto_ranged_put_prepare_request,
     .init_signing_date_time = aws_s3_meta_request_init_signing_date_time_default,
     .sign_request = aws_s3_meta_request_sign_request_default,
-    .incoming_headers = NULL,
-    .incoming_headers_block_done = s_s3_auto_ranged_put_header_block_done,
-    .incoming_body = NULL,
-    .stream_complete = s_s3_auto_ranged_put_stream_complete,
-    .notify_request_destroyed = s_s3_auto_ranged_put_notify_request_destroyed,
+    .finished_request = s_s3_auto_ranged_put_request_finished,
+    .delivered_requests = aws_s3_meta_request_delivered_requests_default,
     .destroy = s_s3_meta_request_auto_ranged_put_destroy,
-    .finish = s_s3_auto_ranged_put_finish,
+    .finish = aws_s3_meta_request_finish_default,
 };
 
-static void s_s3_auto_ranged_put_lock_synced_data(struct aws_s3_auto_ranged_put *auto_ranged_put) {
-    AWS_PRECONDITION(auto_ranged_put);
-
-    aws_mutex_lock(&auto_ranged_put->base.synced_data.lock);
-}
-
-static void s_s3_auto_ranged_put_unlock_synced_data(struct aws_s3_auto_ranged_put *auto_ranged_put) {
-    AWS_PRECONDITION(auto_ranged_put);
-
-    aws_mutex_unlock(&auto_ranged_put->base.synced_data.lock);
-}
-
-static struct aws_s3_meta_request_finish_options *s_copy_finish_options(
```

Comment:
We no longer cache finish options to be used later, so this is no longer needed.